src/aap_eda/api/serializers/activation.py

```diff
@@ -565,9 +565,20 @@ def prepare_update(self, activation: models.Activation):
         else:
             vault_data = VaultData()
         if self.validated_data.get("source_mappings", []):
+            if not rulebook_id:
+                # load the original ruleset
+                self.validated_data[
+                    "rulebook_rulesets"
+                ] = activation.rulebook.rulesets
+
+            # update with new event streams
             self.validated_data[
                 "rulebook_rulesets"
             ] = _update_event_stream_source(self.validated_data, vault_data)
+        elif rulebook_id:
```

Comment:
can we combine this check with https://github.com/ansible/eda-server/pull/1183/files#diff-b75687856dd6a2e0a5e5a1d5d322ed63ca815ecd7fbfbb6aadb25009b4d0e89bL552?